### PR TITLE
fix: Reload version in case of index not found

### PR DESCRIPTION
**Pull request by himank**
## Describe your changes
Reload metadata version in case of search index not found

### CODE REVIEW

File: server/services/v1/search/session.go [hunk 1]
```diff
@@ -64,10 +64,25 @@ func (sessions *SessionManager) Execute(ctx context.Context, runner Runner) (Res
 		return Response{}, errors.NotFound("tenant '%s' not found", namespace)
 	}
 
-	resp, err := runner.Run(ctx, tenant)
+	resp, err := sessions.execute(ctx, tenant, runner)
+	if err != nil && shouldRecheckTenantVersion(err) {
+		_ = sessions.TrackVersion(ctx, tenant)
+
+		resp, err = sessions.execute(ctx, tenant, runner)
```
Comment (efirs) [inline server/services/v1/search/session.go:71]: Can this again return `metadata.ErrCodeProjectNotFound, metadata.ErrCodeSearchIndexNotFound` ?

Comment (himank) [reply server/services/v1/search/session.go:71]: that's right and the `createApiError` at line 74 will wrap it up in user facing error.

**Review by adilansari — DISMISSED** [review]

File: server/services/v1/search/session.go (hunk 1 above)
Comment (efirs) [reply server/services/v1/search/session.go:71]: Then should we run this in the loop, something like:
```
do {
 resp, err := sessions.execute(ctx, tenant, runner)
} while	(err != nil && shouldRecheckTenantVersion(err))
```

Comment (himank) [reply server/services/v1/search/session.go:71]: We only need to check & reload once.

**Review by efirs — APPROVED** [review]

**Review by efirs — APPROVED** [review]

**Pull request merged**